### DISCUSSION TITLE
Differentiate touch input between targets?

### DISCUSSION THREAD

**Discussion by Tyler-Lopez** [patrykandpatrick/vico#860] [Ideas]
### Question

Hi - I know that I can use `markerVisibilityListener` to receive touch input for a given x-value. However, is there a way to differentiate the y-values and which one is closest to what was clicked? I would love if `markerVisibilityListener` included some information about where on the y-axis the touch occurred.

### UI framework(s)

Jetpack Compose

**Comment by Gowsky** [comment]
Hello, @Tyler-Lopez. Just to confirm, is this referring to a case where there are multiple targets associated with a single _x_ value, and you receive all of these targets in `CartesianMarkerVisibilityListener`? Are you looking to determine which of these targets is closest to the touch point in terms of _y_ value?

**Reply by Tyler-Lopez** [reply]
Hi!

Correct: here is a visualization of what I am looking to achieve. This has become a lower priority for me, but I believe that this may not currently be achievable.

![Screenshot 2024-10-04 at 11 26 28 AM](https://github.com/user-attachments/assets/1474d2b2-f87e-474d-ac73-861d1f4f35a4)

Generally speaking, I would also love if the callback from `CartesianMarkerVisibilityListener` did not "lock-onto" X and Y values of the chart for me as a client, but instead was more continuous to touch input. Then, clients could determine the "lock-on" logic themselves and how to respond.

**Reply by Tyler-Lopez** [reply]
As I believe this is not something currently achievable, I've changed this from a "question" to an "idea".

**Reply by Gowsky** [reply]
@Tyler-Lopez, you can achieve this relatively easily by overriding `CartesianChart.drawMarker`, which receives the exact pointer coordinates (without the locking on you described). You can update it so that it considers both _x_ and _y_ values when filtering out the `CartesianMarker.Target`s for which the `CartesianMarker` should be drawn. We'll look into adding a public API that simplifies this kind of customization in the future.
